### PR TITLE
Fix the call to etcdbackup

### LINKED ISSUE #1542: etcdbackup - ContainerCannotRun
```
  containerStatuses:
  - containerID: docker://e622eb245fd82f73b2ffaf9acc541b3c7d2e2fa0d979f8317281d987d9871661
    image: quay.io/openshift-on-azure/azure:latest
    imageID: docker-pullable://quay.io/openshift-on-azure/azure@sha256:74c88242053bb02ca855fb03fb653f9ff0befc23f69ffa7a9f3c12a56e1a0e80
    lastState: {}
    name: etcd-backup
    ready: false
    restartCount: 0
    state:
      terminated:
        containerID: docker://e622eb245fd82f73b2ffaf9acc541b3c7d2e2fa0d979f8317281d987d9871661
        exitCode: 127
        finishedAt: 2019-04-17T22:00:21Z
        message: |
          oci runtime error: container_linux.go:247: starting container process caused "exec: \"etcdbackup\": executable file not found in $PATH"
        reason: ContainerCannotRun
        startedAt: 2019-04-17T22:00:21Z
```

### PR DESCRIPTION
**Pull request by thekad**
* Removed the minimum arg settings in the cobra config
* Renamed calls to the etcdbackup references
* Un-exported the config, not sure why was it exported in the first place

```release-note
NONE
```

Fixes #1542

### CODE REVIEW

File: pkg/entrypoint/etcdbackup/cmd.go [hunk 1]
```diff
@@ -28,16 +27,16 @@ func NewCommand() *cobra.Command {
...
 	cc.Flags().String("destination", "", "Where to place the blob on the filesystem")
 	cc.Flags().Int("maxBackups", 6, "Maximum number of backups to keep")
 	cc.Flags().StringP("action", "a", "save", "Action to be executed [save, download]")
 
 	return cc
 }
 
-func configFromCmd(cmd *cobra.Command) (*Config, error) {
-	c := &Config{}
+func configFromCmd(cmd *cobra.Command) (*commandConfig, error) {
```
Comment (mjudeikis) [inline pkg/entrypoint/etcdbackup/cmd.go]: `Config` is the same struct all around the `cmd` files for all `entrypoint/{entrypoint}/Config` files. So Config kidna fits the name here.
https://github.com/openshift/openshift-azure/blob/master/pkg/entrypoint/metricsbridge/cmd.go#L9
https://github.com/openshift/openshift-azure/blob/master/pkg/entrypoint/startup/cmd.go#L9
etc. Which makes boiler-plate same. Either rename it everywhere to `cmdConfig` or rollback the change.

Comment (thekad) [reply pkg/entrypoint/etcdbackup/cmd.go]: you are right, pushed 6ae2ded0fd76b88e1d9b11368c5ad122dbe78617

**Comment by mjudeikis** [comment]
lgtm, lets wait for tests

**Comment by ehashman** [comment]
Hmm, that didn't seem to work... how do we get the etcdbackup test to run as part of this PR? We should probably gate on that passing here.

/test etcdbackuprecovery

**Comment by mjudeikis** [comment]
@ehashman : 
https://github.com/openshift/release/blob/master/ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-master-presubmits.yaml#L119
 :////

**Comment by ehashman** [comment]
Let's see if it works now that https://github.com/openshift/release/pull/3603 was merged...

/retest
/test e2e-etcdbackuprecovery

**Comment by mjudeikis** [comment]
/retest
/hold
@Makdaam I can see some test are not running:
```
SUCCESS! -- 0 Passed | 0 Failed | 0 Pending | 31 Skipped 
```

**Comment by mjudeikis** [comment]
@thekad  did you tested it manually?

**Comment by jim-minter** [comment]
/hold

I don't see how this can fix #1542.

As well as the MinimumNArgs problem, #1542 is also broken because we are specifying `{"command": ["etcdbackup"], "args": ["--action", "save"]}` in  pkg/sync/v5/data/CronJob.batch/openshift-etcd/etcd-backup.yaml when it should be `{"args": ["etcdbackup", "--action", "save"]}` -- but this is not fixed here.

By the same token the changes to pkg/cluster/kubeclient/backup.go in this PR cannot be right.

Meanwhile, the CI isn't running because I broke it - please could I ask you to fix it by s/EtcdRecovery/EtcdBackupRecovery/ in test/e2e/specs/fakerp/etcdrecovery.go?

**Comment by jim-minter** [comment]
also rename etcdrecovery.go -> etcdbackuprecovery.go

**Comment by thekad** [comment]
> /hold
> 
> I don't see how this can fix #1542.
> 
> As well as the MinimumNArgs problem, #1542 is also broken because we are specifying `{"command": ["etcdbackup"], "args": ["--action", "save"]}` in pkg/sync/v5/data/CronJob.batch/openshift-etcd/etcd-backup.yaml when it should be `{"args": ["etcdbackup", "--action", "save"]}` -- but this is not fixed here.
> 
> By the same token the changes to pkg/cluster/kubeclient/backup.go in this PR cannot be right.
> 
> Meanwhile, the CI isn't running because I broke it - please could I ask you to fix it by s/EtcdRecovery/EtcdBackupRecovery/ in test/e2e/specs/fakerp/etcdrecovery.go?

Ah yes, you are correct, the command is now `azure` since `etcdbackup` is a sub-command now. Updated.

**Comment by asalkeld** [comment]
flake https://github.com/openshift/openshift-azure/issues/1582
/retest

**Pull request merged**